services/headless-lms/models/src/exercise_slide_submissions.rs

```diff
@@ -451,6 +468,100 @@ LIMIT $2 OFFSET $3
     Ok(submissions)
 }
 
+pub async fn exercise_slide_submission_count_with_exam_id(
+    conn: &mut PgConnection,
+    exam_id: Uuid,
+) -> ModelResult<u32> {
+    let count = sqlx::query!(
+        "
+SELECT COUNT(*) as count
+FROM exercise_slide_submissions
+WHERE exam_id = $1
+AND deleted_at IS NULL
+",
+        exam_id,
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(count.count.unwrap_or(0).try_into()?)
+}
+
+pub async fn exercise_slide_submission_count_with_exercise_id(
+    conn: &mut PgConnection,
+    exercise_id: Uuid,
+) -> ModelResult<u32> {
+    let count = sqlx::query!(
+        "
+SELECT COUNT(*) as count
+FROM exercise_slide_submissions
+WHERE exercise_id = $1
+AND deleted_at IS NULL
+",
+        exercise_id,
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(count.count.unwrap_or(0).try_into()?)
+}
+
+pub async fn exercise_slide_submissions_and_user_exercise_state_list_with_exercise_id(
+    conn: &mut PgConnection,
+    exercise_id: Uuid,
+    pagination: Pagination,
+) -> ModelResult<Vec<ExerciseSlideSubmissionAndUserExerciseState>> {
+    let submissions = sqlx::query_as!(
+        ExerciseSlideSubmission,
+        r#"
+        SELECT id,
+        created_at,
+        updated_at,
+        deleted_at,
+        exercise_slide_id,
+        course_id,
+        course_instance_id,
+        exam_id,
+        exercise_id,
+        user_id,
+        user_points_update_strategy AS "user_points_update_strategy: _"
+    FROM exercise_slide_submissions
+    WHERE exercise_id = $1
+      AND deleted_at IS NULL
+      AND created_at in (SELECT MAX(created_at) FROM exercise_slide_submissions GROUP BY user_id, exercise_slide_id)
```

Comment:
Matches all records that have the same created at.

Mayb you can use the SQL HAVING keyword here?